Failure to scan a project with empty properties in the pom.xml

Thanks @[virgium03](https://github.com/virgium03) for raising this issue!

Version: `0.9.1`.

If any of the `pom.xml` files in a project contain empty `properties` elements, the tool fails. The fix is to remove the empty `properties` element from the incriminated `pom.xml` file.

Examples which generate a failure:

`<properties>
</properties>
`

or

```
<properties>
 
</properties>
```


The stacktrace is as follows:

> java.io.UncheckedIOException: Failed to parse C:\Users\ratoico\work\projects\uumds-admin\admin-common\common-facade\pom.xml
        at org.openrewrite.maven.internal.RawMaven.parse(RawMaven.java:90)
        at org.openrewrite.maven.MavenParser.lambda$parseInputs$0(MavenParser.java:74)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at org.openrewrite.maven.MavenParser.parseInputs(MavenParser.java:75)
        at com.vmware.sbm.project.buildfile.RewriteMavenParser.parseInputs(RewriteMavenParser.java:46)
        at org.openrewrite.Parser.parse(Parser.java:39)
        at com.vmware.sbm.project.buildfile.OpenRewriteMavenBuildFilesFactory.applyProjections(OpenRewriteMavenBuildFilesFactory.java:32)
        at com.vmware.sbm.project.SimpleProjectContextBuilder.buildContext(SimpleProjectContextBuilder.java:43)
        at com.vmware.sbm.project.CachingProjectContextBuilder.buildContext(CachingProjectContextBuilder.java:43)
        at com.vmware.sbm.commands.ScanCommand.execute(ScanCommand.java:26)
        at com.vmware.sbm.shell.ScanShellCommand.scan(ScanShellCommand.java:39)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:282)
        at org.springframework.shell.Shell.evaluate(Shell.java:180)
        at org.springframework.shell.Shell.run(Shell.java:142)
        at org.springframework.shell.jline.InteractiveShellApplicationRunner.run(InteractiveShellApplicationRunner.java:84)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:797)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:787)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:333)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1313)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
        at com.vmware.sbm.SpringShellApplication.main(SpringShellApplication.java:10)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:107)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot coerce empty String ("") to element of `java.util.LinkedHashMap<java.lang.String,java.lang.String>` (but could if coercion was enabled using `CoercionConfig`)
 at [Source: (sun.nio.ch.ChannelInputStream); line: 17, column: 5] (through reference chain: org.openrewrite.maven.internal.RawPom["properties"])
        at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67)
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadCoercion(DeserializationContext.java:1666)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._checkCoercionFail(StdDeserializer.java:1432)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromEmptyString(StdDeserializer.java:325)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromString(StdDeserializer.java:276)
        at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:444)
        at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:32)
        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:565)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:449)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:362)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:195)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4593)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3585)
        at org.openrewrite.maven.internal.RawMaven.parse(RawMaven.java:84)
        ... 38 more

--- a/components/sbm-core/src/test/java/org/springframework/sbm/project/buildfile/OpenRewriteMavenBuildFileTest.java
+++ b/components/sbm-core/src/test/java/org/springframework/sbm/project/buildfile/OpenRewriteMavenBuildFileTest.java
@@ -2656,26 +2656,27 @@ public class OpenRewriteMavenBuildFileTest {
 
     @Test
     void upgradeParentVersion() {
-        String pomXml =
-                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
-                        "<project xmlns=\"http://maven.apache.org/POM/4.0.0\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
-                        "         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd\">\n" +
-                        "    <modelVersion>4.0.0</modelVersion>\n" +
-                        "    <parent>\n" +
-                        "        <groupId>org.springframework.boot</groupId>\n" +
-                        "        <artifactId>spring-boot-starter-parent</artifactId>\n" +
-                        "        <version>2.4.12</version>\n" +
-                        "        <relativePath/> <!-- lookup parent from repository -->\n" +
-                        "    </parent>\n" +
-                        "    <groupId>com.example</groupId>\n" +
-                        "    <artifactId>spring-boot-24-to-25-example</artifactId>\n" +
-                        "    <version>0.0.1-SNAPSHOT</version>\n" +
-                        "    <name>spring-boot-2.4-to-2.5-example</name>\n" +
-                        "    <description>spring-boot-2.4-to-2.5-example</description>\n" +
-                        "    <properties>\n" +
-                        "        <java.version>11</java.version>\n" +
-                        "    </properties>\n" +
-                        "</project>\n";
+        String pomXml = """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                    <modelVersion>4.0.0</modelVersion>
+                    <parent>
+                        <groupId>org.springframework.boot</groupId>
+                        <artifactId>spring-boot-starter-parent</artifactId>
+                        <version>2.4.12</version>
+                        <relativePath/> <!-- lookup parent from repository -->
+                    </parent>
+                    <groupId>com.example</groupId>
+                    <artifactId>spring-boot-24-to-25-example</artifactId>
+                    <version>0.0.1-SNAPSHOT</version>
+                    <name>spring-boot-2.4-to-2.5-example</name>
+                    <description>spring-boot-2.4-to-2.5-example</description>
+                    <properties>
+                        <java.version>11</java.version>
+                    </properties>
+                </project>
+                """;
 
         BuildFile openRewriteMavenBuildFile = TestProjectContext.buildProjectContext().withMavenRootBuildFileSource(pomXml).build().getBuildFile();
 

--- a/components/sbm-core/src/test/java/org/springframework/sbm/project/buildfile/OpenRewriteMavenBuildFileTest.java
+++ b/components/sbm-core/src/test/java/org/springframework/sbm/project/buildfile/OpenRewriteMavenBuildFileTest.java
@@ -21,6 +21,7 @@ import org.junit.jupiter.api.*;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.sbm.GitHubIssue;
 import org.springframework.sbm.build.api.BuildFile;
 import org.springframework.sbm.build.api.DependenciesChangedEvent;
 import org.springframework.sbm.build.api.Dependency;
@@ -2682,5 +2683,26 @@ public class OpenRewriteMavenBuildFileTest {
 
         assertThat(openRewriteMavenBuildFile.getParentPomDeclaration()).isNotEmpty();
         assertThat(openRewriteMavenBuildFile.getParentPomDeclaration().get().getVersion()).isEqualTo("2.5.6");
+    }
+
+    @GitHubIssue("https://github.com/spring-projects-experimental/spring-boot-migrator/issues/55")
+    @Test
+    void parsingPomWithEmptyPropertiesSectionShouldSucceed() {
+        String pomXml = """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                    <modelVersion>4.0.0</modelVersion>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-starter-parent</artifactId>
+                    <version>2.4.12</version>
+                    <properties>
+                    
+                    </properties>
+                </project>
+                """;
+
+        ProjectContext projectContext = TestProjectContext.buildProjectContext().withMavenRootBuildFileSource(pomXml).build();
+        assertThat(projectContext.getApplicationModules().getRootModule().getBuildFile()).isNotNull();
     }
 }

